core/common/src/main/java/alluxio/uri/Authority.java

```diff
@@ -48,7 +48,7 @@ static Authority fromString(String authority) {
       } else {
         matcher = MULTI_MASTERS_AUTH.matcher(authority);
         if (matcher.matches()) {
-          return new MultiMasterAuthority(authority);
+          return new MultiMasterAuthority(authority.replaceAll("[;+]", ","));
```

Comment:
Put the replacement here instead of in `MultiMasterAuthority` constructor to fix the error in 
`alluxio.client.file.BaseFileSystem.checkUri()`. If we put it in the `MultiMasterAuthority` constructor, checkUri will show that:

```
2019-01-07 04:53:05,282 ERROR AbstractShell - Error running ls -R alluxio://masters-1:19998+masters-2:19998/results
java.lang.IllegalArgumentException: The URI authority masters-1:19998+masters-2:19998 does not match the configured value of masters-1:19998,masters-2:19998.
	at alluxio.client.file.BaseFileSystem.checkUri(BaseFileSystem.java:577)
```

Reply:
lgtm